hera_cal/firstcal.py

```diff
@@ -397,26 +397,29 @@ def UVData_to_dict(uvdata_list, filetype='miriad'):
             uv_in = UVData()
             # read in file without multiple if statements
             getattr(uv_in, 'read_' + filetype)(fname)
-        # reshape data and flag arrays to make slicing time and baselines easy
-        data = uv_in.data_array.reshape(
-            uv_in.Ntimes, uv_in.Nbls, uv_in.Nspws, uv_in.Nfreqs, uv_in.Npols)
-        flags = uv_in.flag_array.reshape(
-            uv_in.Ntimes, uv_in.Nbls, uv_in.Nspws, uv_in.Nfreqs, uv_in.Npols)
 
-        for nbl, (i, j) in enumerate(map(uv_in.baseline_to_antnums, uv_in.baseline_array[:uv_in.Nbls])):
+        # iterate over unique baselines
+        for nbl, (i, j) in enumerate(map(uv_in.baseline_to_antnums, np.unique(uv_in.baseline_array))):
             if (i, j) not in d:
                 d[i, j] = {}
                 f[i, j] = {}
             for ip, pol in enumerate(uv_in.polarization_array):
                 pol = pol2str[pol]
+                try:
```

Comment:
we can get rid of the try except by using uv_in.get_data(i, j, pol).

Reply:
ah, cool, didn't even know you could do that..